src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -117,6 +120,57 @@ public void SetUp()
                 mockSonarLanguageRecognizer.Object, mockAnalysisRequester.Object, mockTaggableBufferIndicator.Object, issueConsumerFactory.Object, logger, mockAnalysisScheduler.Object, new NoOpThreadHandler());
         }
 
+
+        #region MEF tests
+
+        [TestMethod]
+        public void CheckIsSingletonMefComponent()
+            => MefTestHelpers.CheckIsSingletonMefComponent<TaggerProvider>();
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported_TaggerProvider()
+            => MefTestHelpers.CheckTypeCanBeImported<TaggerProvider, ITaggerProvider>(GetRequiredExports());
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported_DocumentEvents()
+            => MefTestHelpers.CheckTypeCanBeImported<TaggerProvider, IDocumentEvents>(GetRequiredExports());
+
+        [TestMethod]
+        public void MefCtor_Check_SameInstanceExported()
```

Comment:
... and a test that the exports return the same instance, which is what we need to happen.

This is based on the [equivalent test](https://github.com/SonarSource/sonarlint-visualstudio/blob/7543d596f851c0d09b910da7996162d6c781531b/src/ConnectedMode.UnitTests/Suppressions/ServerIssuesStoreTests.cs#L45) for `ServerIssuesStore`.

At the time the `ServerIssuesStoreTest` was written we didn't have the test helper `MefTestHelpers.CheckIsSingletonMefComponent`. In theory we could just rely on the fact that the export is marked as a singleton and assume we'd get the same instance for each export. However, this test is more explicit and not particularly complicated.

If we had more types with multiple exports it would be worth adding a new test helper.